operators/src/opencl/cl_program.rs

```diff
@@ -404,31 +429,61 @@ impl<'a> CLProgramRunnable<'a> {
                         // no geo
                     }
                     TypedFeatureCollection::MultiPoint(points) => {
-                        let coordinates = points.coordinates();
-                        let buffer = Buffer::<Double2>::builder()
-                            .queue(queue.clone())
-                            .len(coordinates.len())
-                            .copy_host_slice(unsafe {
-                                std::slice::from_raw_parts(
-                                    coordinates.as_ptr() as *const Double2,
-                                    coordinates.len(),
-                                )
-                            })
-                            .build()?;
-
-                        kernel.set_arg(format!("IN_POINT_COORDS{}", idx), &buffer)?;
-
-                        let coordinates_offsets = points.multipoint_offsets();
-                        let buffer = Buffer::builder()
-                            .queue(queue.clone())
-                            .len(coordinates_offsets.len())
-                            .copy_host_slice(coordinates_offsets)
-                            .build()?;
-
-                        kernel.set_arg(format!("IN_POINT_OFFSETS{}", idx), &buffer)?;
+                        kernel.set_arg(
+                            format!("IN_FEATURE{}_COORDS", idx),
+                            &create_coordinate_buffer(queue, points.coordinates())?,
+                        )?;
+
+                        kernel.set_arg(
+                            format!("IN_FEATURE{}_OFFSETS", idx),
+                            &create_offset_buffer(queue, points.multipoint_offsets())?,
+                        )?;
+
+                        kernel.set_arg(format!("IN_FEATURE{}_LEN", idx), points.len() as i32)?;
```

Comment:
don't you need offsets_len and coords_len?

Reply:
also maybe calll it num_features? because in_feature{} is not an array so why does it have len?

Reply:
offsets_len is basically the same as num_features but for lines and polygons there are multiple offset arrays and each array needs a len doesn't it?

Reply:
One is enough to get the other length from within the kernel, cf. my test kernels. I will reconsider the naming.

Reply:
b9b625f